src/iuse.cpp

```diff
@@ -1841,6 +1841,32 @@ int iuse::purify_iv(player *p, item *it, bool, const tripoint& )
     return it->type->charges_to_use();
 }
 
+void spawn_spores( const player &p ) {
+    int spores_spawned = 0;
+    for ( const tripoint &dest : closest_tripoints_first( 4, p.pos() ) ) {
+        if ( g->m.move_cost( dest ) == 0 ) { // impassable terrain
+            continue;
+        }
+        float dist = trig_dist( dest, p.pos() );
+        if ( x_in_y( 1, dist ) ) { // slightly less frequent than previous code
+            g->m.marlossify( dest );
+        }
+        if ( dest == p.pos() ) { // on top of the player
+            continue;
+        }
+        if ( g->mon_at( dest ) != -1 ) { // there's already a monster there
```

Comment:
I've been working on Cataclysm for months and I still don't intuitively recognize that zero move cost means impassable terrain. I always first think of it as the cheapest/fastest movement.

I'll take all the rest of the advice though.